File: Python/done/1_Two_Sum/main2.js
const nums = [2, 7, 11, 15];

const target = 9;

const main = (nums, target) => {
  const map = new Map();

  for (let i = 0; i < nums.length; i++) {
    const complement = target - nums[i];
    if (map.has(complement)) {
      return [map.get(complement), i];
    } else {
      map.set(nums[i], i);
    }
  }

  return [];
};

console.log(main(nums, target));

File: Python/done/1_Two_Sum/main2.py
def main(nums, target):
    map = {}
    for i, num in enumerate(nums):
        complement = target - num
        if complement in map:
            return [map[complement], i]
        map[num] = i

    return []


nums = [2, 7, 11, 15]
target = 9
result = main(nums, target)
print(result)
